From 9d7568b9e62bf4268998238152b0e4246ecf1528 Mon Sep 17 00:00:00 2001
From: Michael J. Ruhl <mjruhl@habana.ai>
Date: Wed, 28 May 2025 20:03:49 +0300
Subject: [PATCH] [SW-227154] g3: increase graceful reset timeout

The graceful reset notifies the LKD that a reset is
eminent, waits for a bit to allow the LKD time to get
ready, and then resets the FW.

It has been determined that if the NIC HW is configured,
the current 2 second timeout is not sufficient.

The recommended timeout is 5 seconds. Since this was
measured with a current configuration, increase the timeout
to 7 seconds to support this and future configs.

Change-Id: Ib1e0665496e46b710519b1175239f686d10ce147
Signed-off-by: Michael J. Ruhl <mjruhl@habana.ai>
---

diff --git a/arch/common/gaudi/common.c b/arch/common/gaudi/common.c
index aeba4c6..8055a90 100644
--- a/arch/common/gaudi/common.c
+++ b/arch/common/gaudi/common.c
@@ -339,9 +339,15 @@
 #define RESET_DELAY_MS		1000
 
 /* Delay secured device reset after notifying the driver in order to give
- * a chance for a graceful reset
+ * a chance for a graceful reset.
+ * NOTE: The FW notifies the LKD of the pending reset, waits, assumes the LKD
+ * is ready, and then does the reset.  The longest pole appears to be the NIC
+ * shutdown path.  It was suggested that 5 seconds could be the right amount of
+ * time to wait, but that it was dependent on the NIC configuration, so 7 seconds
+ * is going be used.
+ *
  */
-#define SECURE_RESET_DELAY_MS	2000
+#define SECURE_RESET_DELAY_MS	7000
 
 EMPTY_WEAK_FUNC(void hl_dev_reset(struct hl_dev_config *dev_config),)
 
